source/val/validate_memory.cpp

```diff
@@ -1017,7 +1146,20 @@ spv_result_t ValidateCopyMemory(ValidationState_t& _, const Instruction* inst) {
 
     if (auto error = CheckMemoryAccess(_, inst, 3)) return error;
   }
-  return ValidateCopyMemoryMemoryAccess(_, inst);
+  if (auto error = ValidateCopyMemoryMemoryAccess(_, inst)) return error;
+
+  // Get past the pointers to avoid checking a pointer copy.
+  auto sub_type = _.FindDef(target_pointer_type->GetOperandAs<uint32_t>(2));
+  while (sub_type->opcode() == SpvOpTypePointer) {
```

Comment:
Arguable this bit of logic -- to trace down past pointers -- could be a common method.  But no request to change this now.